File: src/lambda/dms-switch/utils/get-change-set.ts
import { Change, CloudFormationClient, DescribeChangeSetCommand } from "@aws-sdk/client-cloudformation";

export const getChangeSet = async ({
  cf,
  changes = [],
  ChangeSetName,
  NextToken,
  StackName,
}: {
  cf: CloudFormationClient;
  StackName: string;
  ChangeSetName: string;
  changes: Change[];
  NextToken?: string;
}): Promise<Change[]> => {
  const changeSetCommand = new DescribeChangeSetCommand({
    StackName,
    ChangeSetName,
    NextToken,
  });
  const changeSet = await cf.send(changeSetCommand);
  const nextChanges = [...changes, ...(changeSet.Changes || [])];
  if (changeSet.NextToken) {
    return getChangeSet({ cf, StackName, ChangeSetName, changes: nextChanges, NextToken: changeSet.NextToken });
  } else {
    return nextChanges;
  }
};

File: src/lambda/dms-switch-py/utils/get_change_set.py

def get_change_set(cf, stack_name, change_set_name, changes=None, next_token=None):
    if changes is None:
        changes = []
    response = cf.describe_change_set(
        StackName=stack_name,
        ChangeSetName=change_set_name,
        NextToken=next_token
    )
    change_set = response.get('Changes', [])
    next_changes = changes + change_set
    if 'NextToken' in response:
        return get_change_set(cf, stack_name, change_set_name, next_changes, response['NextToken'])
    else:
        return next_changes
